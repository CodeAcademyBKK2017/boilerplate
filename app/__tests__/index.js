import 'react-native';
import App from '../index';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />
    );
    expect(tree).toBeDefined();
  });
});
it('Check Function', () => { // example to test class methods
  const props = {};
  const wrapper = shallow(<App {...props}/>);
  const instance = wrapper.instance();
  expect(instance.state.text).toEqual('');
  instance.onCount('some');
  expect(instance.state.text).toEqual('some');
});
it('Check Function onTitle', () => { // example to test class methods
  const props = {};
  const wrapper = shallow(<App {...props}/>);
  const instance = wrapper.instance();
  expect(instance.state.title).toEqual('');
  instance.onTitle('some');
  expect(instance.state.title).toEqual('some');
});
it('Check Function saveNote', () => {
  const props = {};
  const wrapper = shallow(<App {...props}/>);
  const instance = wrapper.instance();
  expect(instance.state.title).toEqual('');
  instance.saveNote('');
  expect(instance.state.title).toEqual('');
});
it('Check Function _keyExtractor', () => {
  const props = {};
  const wrapper = shallow(<App {...props}/>);
  const instance = wrapper.instance();
  expect(instance.state.title).toEqual('');
  instance._keyExtractor('12345');
  expect(instance.state.title).toEqual('');
});
it('rendersItem correctly', () => {
  const wrapper = shallow(<App/>);
  const instance = wrapper.instance();
  const item = {title: 'title', uuid: '1'};
  const snapshot = instance._renderItem({item});
  expect(snapshot).toMatchSnapshot();
});