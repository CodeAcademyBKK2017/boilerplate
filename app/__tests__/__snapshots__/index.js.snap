// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendersItem correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "padding": 3,
    }
  }
>
  title
</Text>
`;
